File: backend/src/application/schemas/candidate.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional, List

class CandidateSummaryDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    full_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    current_status: str
    profile_quality: float
    latest_workflow_status: Optional[str] = None
    latest_extraction_confidence: Optional[float] = None
    created_at: datetime
    updated_at: datetime

class CandidateSkillDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    name: str
    proficiency: Optional[str] = None

class CandidateExperienceDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    company: str
    title: str
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    description: Optional[str] = None

class CandidateEducationDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    institution: str
    degree: Optional[str] = None
    field_of_study: Optional[str] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    description: Optional[str] = None

class CandidateProjectDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    name: str
    description: Optional[str] = None
    url: Optional[str] = None

class CandidateProfileDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    status: str
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    summary: Optional[str] = None
    created_at: datetime
    updated_at: datetime

class CandidateDetailsDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    profile: CandidateProfileDTO
    skills: List[CandidateSkillDTO] = []
    experience: List[CandidateExperienceDTO] = []
    education: List[CandidateEducationDTO] = []
    projects: List[CandidateProjectDTO] = []

File: frontend/src/types/candidate.ts
import { z } from "zod";
import { PaginatedResponse } from "./pagination";

export const CandidateSummarySchema = z.object({
  id: z.string().uuid(),
  full_name: z.string(),
  email: z.string().email().nullable().optional(),
  phone: z.string().nullable().optional(),
  current_status: z.string(),
  profile_quality: z.number(),
  latest_workflow_status: z.string().nullable().optional(),
  latest_extraction_confidence: z.number().nullable().optional(),
  created_at: z.string().datetime(),
  updated_at: z.string().datetime(),
});

export const CandidateSkillSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
  proficiency: z.string().nullable().optional(),
});

export const CandidateExperienceSchema = z.object({
  id: z.string().uuid(),
  company: z.string(),
  title: z.string(),
  start_date: z.string().datetime().nullable().optional(),
  end_date: z.string().datetime().nullable().optional(),
  description: z.string().nullable().optional(),
});

export const CandidateEducationSchema = z.object({
  id: z.string().uuid(),
  institution: z.string(),
  degree: z.string().nullable().optional(),
  field_of_study: z.string().nullable().optional(),
  start_date: z.string().datetime().nullable().optional(),
  end_date: z.string().datetime().nullable().optional(),
  description: z.string().nullable().optional(),
});

export const CandidateProjectSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
  description: z.string().nullable().optional(),
  url: z.string().url().nullable().optional(),
});

export const CandidateProfileSchema = z.object({
  id: z.string().uuid(),
  status: z.string(),
  first_name: z.string().nullable().optional(),
  last_name: z.string().nullable().optional(),
  email: z.string().email().nullable().optional(),
  phone: z.string().nullable().optional(),
  summary: z.string().nullable().optional(),
  created_at: z.string().datetime(),
  updated_at: z.string().datetime(),
});

export const CandidateDetailsSchema = z.object({
  profile: CandidateProfileSchema,
  skills: z.array(CandidateSkillSchema).default([]),
  experience: z.array(CandidateExperienceSchema).default([]),
  education: z.array(CandidateEducationSchema).default([]),
  projects: z.array(CandidateProjectSchema).default([]),
});

export type CandidateSummaryDTO = z.infer<typeof CandidateSummarySchema>;
export type CandidateSkillDTO = z.infer<typeof CandidateSkillSchema>;
export type CandidateExperienceDTO = z.infer<typeof CandidateExperienceSchema>;
export type CandidateEducationDTO = z.infer<typeof CandidateEducationSchema>;
export type CandidateProjectDTO = z.infer<typeof CandidateProjectSchema>;
export type CandidateProfileDTO = z.infer<typeof CandidateProfileSchema>;
export type CandidateDetailsDTO = z.infer<typeof CandidateDetailsSchema>;

export type CandidateListResponse = PaginatedResponse<CandidateSummaryDTO>;
